### PR TITLE
[MAINT] Finish adding tests for `_utils.data_gen`

### LINKED ISSUE #2750: Improve testing of `_utils.data_gen`
It looks like some functions of this module are either not tested correctly, or not tested at all.
For example: `generate_labeled_regions_large`, `write_fake_bold_img`, `generate_signals_from_precisions`....
I think it would help to have all these functions unit-tested in the same file (`nilearn/tests/test_testing.py` for example or some other file?).

### PR DESCRIPTION
**Pull request by kimsin98**
- Closes #2750

Changes proposed in this pull request:
- Add tests for public data_gen functions
- Fix 2 data_gen functions to better match their doc

### CODE REVIEW

File: nilearn/_utils/data_gen.py [hunk 1]
```diff
@@ -649,7 +644,7 @@ def generate_group_sparse_gaussian_graphs(
         Number of signals per subject to generate.
 
     min_n_samples, max_n_samples : :obj:`int`, optional
-        Each subject have a different number of samples, between these two
+        Each subject have a random number of samples, between these two
```
Comment (bthirion) [inline nilearn/_utils/data_gen.py:647]:
```suggestion
        Each subject has a random number of samples, between these two
```

File: nilearn/_utils/tests/test_data_gen.py [hunk 2]
```diff
@@ -531,3 +571,119 @@ def test_generate_fake_fmri_error(rng):
             block_size=None,
             random_state=rng,
         )
+
+
+@pytest.mark.parametrize(
+    "shapes", [[(2, 3, 5, 7)], [(5, 5, 5, 3), (5, 5, 5, 5)]]
```
Comment (bthirion) [inline nilearn/_utils/tests/test_data_gen.py:577]: Would it be possible to consider only two cases ? The point is to keep test time under control.

**Review by bthirion — COMMENTED** [review]
LGTM overall, thx !

File: nilearn/_utils/tests/test_data_gen.py (hunk 2 above)
Comment (kimsin98) [reply nilearn/_utils/tests/test_data_gen.py:577]: Do you mean 2 cases per test or per parameter? Should other tests also reduce them to 2?

Comment (bthirion) [reply nilearn/_utils/tests/test_data_gen.py:577]: per parameter, because the number of combinations quickly explodes.

Comment (kimsin98) [reply nilearn/_utils/tests/test_data_gen.py:583]: Oh, then this should be ok. There are 2 lists tested here. I reduced parameters for `generate_fake_fmri` though. Most of the cases are skipped when `n_block` is None anyway.

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by bthirion** [comment]
Thx. This makes a noticeable coverage increase. Do you know how much this increases test time ?

**Comment by Remi-Gau** [comment]
> Thx. This makes a noticeable coverage increase. Do you know how much this increases test time ?

This is the run time of the test suite since december: can be downloaded and browsed from this [artefact](https://github.com/nilearn/nilearn/actions/runs/9830297425/artifacts/1675542643)

newplot

The run time I am seeing on the test suite on this PR do not seem to induce a big increase in terms of time: am fairly confident that the tested functions are actually pretty quick to run.

https://github.com/nilearn/nilearn/actions/runs/9803927481

**Comment by bthirion** [comment]
All right, then !

**Comment by Remi-Gau** [comment]
@kimsin98 

I will add a changelog entry and your name as a contributor.

File: CITATION.cff [hunk 3]
```diff
@@ -652,6 +652,11 @@ authors:
     orcid: https://orcid.org/0000-0002-6437-0700
     affiliation: Danish Research Centre for Magnetic Resonance, Copenhagen University
       Hospital Amager & Hvidovre, Copenhagen, Denmark.
+  - given-names: Sin
+    family-names: Kim
+    website: https://github.com/kimsin98
+    orcid: https://orcid.org/0000-0003-4652-3758
+    affiliation: 'Korea Advanced Institute of Science and Technology: Daejeon, KR'
```
Comment (Remi-Gau) [inline CITATION.cff:659]: @kimsin98 

I got those from your github profile, I hope they are correct.

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
Will merge but @kimsin98 feel free to update your info in the citation.cff at any time.

**Pull request merged**